iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -1125,6 +1125,43 @@ def HAL_CommandBufferDeviceOp : HAL_PureOp<"command_buffer.device"> {
   let hasCanonicalizer = 1;
 }
 
+def HAL_CommandBufferBeginDebugGroupOp : HAL_Op<"command_buffer.begin_debug_group"> {
+  let summary = [{pushes a command buffer debug group label}];
+  let description = [{
+    Pushes a new debug group with the given label.
+    All commands between this and a mandatory matching call to
+    `hal.command_buffer.end_debug_group` will be grouped together with the
+    given label.
```

Comment:
It would be nice if a verifier (or verification pass) to check for this was easy to add and lightweight to run.